pkg/conversion/mapi2capi/util.go

```diff
@@ -17,72 +17,163 @@ limitations under the License.
 package mapi2capi
 
 import (
-	mapiv1beta1 "github.com/openshift/api/machine/v1beta1"
+	"fmt"
 
-	corev1 "k8s.io/api/core/v1"
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
-// fromMAPIMachineToCAPIMachine translates a MAPI Machine to its Core CAPI Machine correspondent.
-func fromMAPIMachineToCAPIMachine(mapiMachine *mapiv1beta1.Machine, apiVersion, kind string) (*capiv1.Machine, field.ErrorList) {
-	var errs field.ErrorList
+func convertMAPIMachineSetSelectorToCAPI(mapiSelector metav1.LabelSelector) metav1.LabelSelector {
+	capiSelector := mapiSelector
+	capiSelector.MatchLabels = convertMAPILabelsToCAPI(mapiSelector.MatchLabels)
+
+	return capiSelector
+}
+
+func convertMAPILabelsToCAPI(mapiLabels map[string]string) map[string]string {
+	capiLabels := make(map[string]string)
 
-	capiMachine := &capiv1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            mapiMachine.Name,
-			Namespace:       capiNamespace,
-			Labels:          mapiMachine.Labels,
-			Annotations:     mapiMachine.Annotations,
-			OwnerReferences: nil, // OwnerReferences not populated here. They are added later by the machineSync controller.
+	toConvertLabels := map[string]string{
```

Comment:
What's the difference between convert and transform?

Reply:
Not much at the moment, I'll drop the toConvert.